src/python/grpcio_reflection/grpc_reflection/v1alpha/proto_reflection_descriptor_database.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2022 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Reference implementation for reflection client in gRPC Python."""
+
+from google.protobuf.descriptor_database import DescriptorDatabase
+from google.protobuf.descriptor_pb2 import FileDescriptorProto
+import grpc
+from grpc_reflection.v1alpha.reflection_pb2 import (
+    ErrorResponse,
+    ExtensionNumberResponse,
+    ExtensionRequest,
+    FileDescriptorResponse,
+    ListServiceResponse,
+    ServerReflectionRequest,
+    ServerReflectionResponse,
+    ServiceResponse,
+)
+from grpc_reflection.v1alpha.reflection_pb2_grpc import ServerReflectionStub
+import logging
+from typing import Iterable, Any
+
+
+class ProtoReflectionDescriptorDatabase(DescriptorDatabase):
+    """
+    ProtoReflectionDescriptorDatabase takes a stub of ServerReflection and provides the methods defined by
+    DescriptorDatabase interfaces. It can be used to feed a DescriptorPool instance.
```

Comment:
As Google Python Style Guide mentioned: https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings

> A docstring should be organized as a summary line (one physical line not exceeding 80 characters) terminated by a period, question mark, or exclamation point. When writing more (encouraged), this must be followed by a blank line, followed by the rest of the docstring starting at the same cursor position as the first quote of the first line. There are more formatting guidelines for docstrings below.

nit: let's add a summary line

Reply:
Redid the docstrings for the whole file + sphinx doc